### ISSUE TITLE
stale UI values after update

### ISSUE THREAD

**Issue by ainesophaur** [filamentphp/filament#2797]
### Package

filament/spatie-tags-plugin

### Package Version

^2.13.4

### Laravel Version

v9.17.0

### Livewire Version

v2.10.5

### PHP Version

PHP 8.1.4

### Bug description

When updating a form and remaining on the page, the tags input field will either retain removed values or will remove newly added values from the UI itself. The database receives the correct values and refreshes to the page will show the correct values. Currently using `spatie/laravel-tags` 4.3.2

I have tried with saving the relationship manually, and that causes the db to lose the correct values after `update()` was called. Also tried a combination of including `tags` in my form fill or leaving it empty/null

Attached are two screen caps, one for removing a tag and one for adding tags

Removing tag with manual refresh after

![tags_issue](https://user-images.githubusercontent.com/4686198/174135319-0c8cf329-6549-463d-98d2-8033a1b9bfdf.gif)

Adding tag with manual refresh after

![tags_add_issue](https://user-images.githubusercontent.com/4686198/174135354-2768619c-2cc8-4430-bfd4-c8efec5922d9.gif)

### Steps to reproduce

```php
use Filament\Forms\Components\SpatieTagsInput;

class EditDemoComponent extends Component implements Forms\Contracts\HasForms {

use Forms\Concerns\InteractsWithForms;

    public $tags;
    public ModelWithHasTags $modelWithHasTags;

    protected function getFormSchema(): array
    {
        return [
            SpatieTagsInput::make('tags')
                ->type('support'),
        ];
    }
    
    protected function getFormModel(): ModelWithHasTags
    {
        return $this->modelWithHasTags;
    }    

    public function submit()
    {
        $this->modelWithHasTags->update($this->form->getState());
    }
}
```

**Comment by danharrin** [comment]
Could you add the following and try again please, the rest looks correct:

```php
public function mount(): void
{
    $this->form->fill();
}
```

**Comment by ainesophaur** [comment]
@danharrin adding the above mount function fixed the issue. Now I only see the call to livewire to set the option and then another call to fetch the available options. 

I know the documentation mentions `filament/tables` use `filament/forms` in the background, but I dont see any specifics in the documentation about adding a mount function (unlike the documentation for `filament/forms`.) I was curious if I should be adding the above method to all of my tables or if its only something to do for this issue.

**Comment by danharrin** [comment]
The form docs should hopefully make it very clear that calling `form->fill()` is **mandatory** for every form to be configured properly.

**Comment by ainesophaur** [comment]
:( I apologize as I was thinking I was updating #2815 (which I prematurely updated as the issue for #2815 so disregard my previous update)

I didn't include a mount in my sample code above, but the actual code from the screen cap does include a mount function to fill in the form

```php
public function mount(): void
    {      
        $this->form->fill([
            'ticket_board_id' => $this->ticket->ticket_board_id,
            'ticket_status_id' => $this->ticket->ticket_status_id,
            'ticket_priority_id' => $this->ticket->ticket_priority_id,
            'ticket_assigned_to_id' => $this->ticket->assigned_to_id,
        ]);

    }
```

**Comment by danharrin** [comment]
It looks like there are styling issues too, do you have console errors?

**Comment by ainesophaur** [comment]
The styling comes from overrides we're doing in CSS to match the rest of the inputs on the site. No console errors reported. 

The only way I've been able to get around the issue is with another call to `$form->fill()` after calling `$model->refresh()` but it causes a UI blip between `newState` -> `oldState` -> `newState`

for example, my form submit calls a function which then emits an event, listened to by the form component

`$this->emit("ticket." . $this->ticket->hashed_id . ".details.updated");`

```php

...
public function handleTicketDetailsUpdated()
    {
        $this->ticket->refresh();
        $this->form->fill([
            'ticket_board_id' => $this->ticket->ticket_board_id,
            'ticket_status_id' => $this->ticket->ticket_status_id,
            'ticket_priority_id' => $this->ticket->ticket_priority_id,
            'ticket_assigned_to_id' => $this->ticket->assigned_to_id,
            'tags' => $this->ticket->tagsWithType('support')->pluck('name')->toArray(),
        ]);
    }
```

If it helps, I can create an actual sample project (minus styling changes etc) and provide you with a repo URL.

**Comment by ainesophaur** [comment]
Ive created a sample sail based project which replicates the issue. Only prep required is to run `php artisan migrate --seed`

https://github.com/ainesophaur/filament-tags-issue-2797

**Comment by danharrin** [comment]
Fixed by #2999. Stupid tiny bug ahah
